lib/inspec/schema/schema.rb

```diff
@@ -0,0 +1,64 @@
+require "json"
+require "lib/inspec/schema/primitives"
+require "lib/inspec/schema/exec_json"
+require "lib/inspec/schema/exec_json_min"
+require "lib/inspec/schema/profile_json"
+
+module Inspec
+  module OutputSchema
+    # Build our definitions
+    def self.build_definitions(schema_type)
+      # Init definition hash
+      definitions = {}
+
+      # Fetch all dependencies
+      defined_types = schema_type.all_depends
+
+      # Add them all to the definition hash
+      defined_types.each do |type|
+        definitions[type.name] = type.body
+      end
+
+      # Nest under "definitions" label
+      {
+        "definitions" => definitions,
+      }
```

Comment:
All of the comments are redundant distractions. Each "paragraph" of code is separated but actually all related. And all boil down to something that's both succinct and easily comprehensible:

```ruby
{
  "definitions" => schema_type.all_depends.map { |t| [t.name, t.body] }.to_h
}
```

Reply:
Used your change in https://github.com/inspec/inspec/pull/4330/commits/7fe53fc1fc53ba04e5579e5542011d7d5f61bba1